src/core/gui/inputdevices/PenInputHandler.cpp

```diff
@@ -171,46 +179,57 @@ auto PenInputHandler::actionStart(InputEvent const& event) -> bool {
     return true;
 }
 
-double PenInputHandler::inferPressureIfEnabled(PositionInputData const& pos, XojPageView* page) {
-    if (pos.pressure == Point::NO_PRESSURE && this->inputContext->getSettings()->isPressureGuessingEnabled()) {
-        PositionInputData lastPos = getInputDataRelativeToCurrentPage(page, this->lastEvent);
-
-        double dt = (pos.timestamp - lastPos.timestamp) / 10.0;
-        double distance = utl::Point<double>(pos.x, pos.y).distance(utl::Point<double>(lastPos.x, lastPos.y));
-        double inverseSpeed = dt / (distance + 0.001);
-
-        // This doesn't have to be exact. Arctan is used here for its sigmoid-like shape,
-        // so that lim inverseSpeed->infinity (newPressure) is some finite value.
-        double newPressure = 3.142 / 2.0 + std::atan(inverseSpeed * 3.14 - 1.3);
+double PenInputHandler::inferPressureValue(PositionInputData const& pos, XojPageView* page) {
```

Comment:
Didn't we have a paper or something for this algorithm? It would be good, to have the reference in the code.